test/auth_test.go

```diff
@@ -182,6 +243,40 @@ func TestTokenAuth(t *testing.T) {
 	nc.Close()
 }
 
+func TestTokenHandlerAuth(t *testing.T) {
+	opts := test.DefaultTestOptions
+	opts.Port = 8232
+	secret := "S3Cr3T0k3n!"
+	opts.Authorization = secret
+	s := RunServerWithOptions(opts)
+	defer s.Shutdown()
+
+	_, err := nats.Connect("nats://localhost:8232")
+	if err == nil {
+		t.Fatal("Should have received an error while trying to connect")
+	}
+
+	tokenURL := fmt.Sprintf("nats://%s@localhost:8232", secret)
+	nc, err := nats.Connect(tokenURL)
+	if err != nil {
+		t.Fatal("Should have connected successfully")
+	}
+	nc.Close()
+
+	// Use Options
+	nc, err = nats.Connect("nats://localhost:8232", nats.TokenHandler(func() string { return secret }))
+	if err != nil {
+		t.Fatalf("Should have connected successfully: %v", err)
+	}
+	nc.Close()
+	// Verify that token in the URL takes precedence.
+	nc, err = nats.Connect(tokenURL, nats.TokenHandler(func() string { return "badtoken" }))
```

Comment:
I think this should be considered an error. Can only have one IMO..

Reply:
OK. Should `TestTokenAuth` also be changed to return an error for this case?